### PR TITLE
VIP-9 Deploy FEI, DAI, USDC Compound PCV Deposits

### PR DESCRIPTION
**Pull request by ElliotFriedman**

### CODE REVIEW

File: contracts/test/integration/vip/vip9.sol [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            toWhitelist
+        );
+        proposal[0].description = "Pause redemptions on the FEI PSM";
+    }
+
+    function mainnetSetup() public override {}
+
+    function mainnetValidate() public override {
+        // assertTrue(
+        //     PCVGuardian(MainnetAddresses.PCV_GUARDIAN).isWhitelistAddress(pcvDeposit)
```
Comment (thomas-waite) [inline contracts/test/integration/vip/vip9.sol:49]: All of this presumably needs uncommenting

File: proposals/dao/vip_9.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+*/
+
+const vipNumber = '9';
+
+// Do any deployments
+// This should exclusively include new contract deployments
+const deploy: DeployUpgradeFunc = async (deployAddress: string, addresses: NamedAddresses, logging: boolean) => {
+  const daiCompoundPCVDeposit = await (
+    await ethers.getContractFactory('ERC20CompoundPCVDeposit')
```
Comment (thomas-waite) [inline proposals/dao/vip_9.ts:30]: Very minor but I'd probs just get the factory once, store in a local var and reuse:

`const ERC20CompoundDepositFactory = await ethers.getContractFactory('ERC20CompoundPCVDeposit');`

Fine as is, just makes it more DRY

File: contracts/test/integration/vip/vip9.sol [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        returns (TimelockSimulation.action[] memory proposal)
+    {
+        proposal = new TimelockSimulation.action[](1);
+
+        address[] memory toWhitelist = new address[](3);
+        // toWhitelist[0] = address(daiDeposit);
+        // toWhitelist[1] = address(feiDeposit);
+        // toWhitelist[2] = address(usdcDeposit);
+
+        proposal[0].target = MainnetAddresses.PCV_GUARDIAN;
```
Comment (thomas-waite) [inline contracts/test/integration/vip/vip9.sol:36]: This needs updating along with the description, think you're aware

**Review by thomas-waite — COMMENTED** [review]
Looks good overall. The Solidity proposal needs some updates, uncommenting, descriptions updating

File: contracts/test/integration/vip/vip9.sol (hunk 1 above)
Comment (ElliotFriedman) [reply contracts/test/integration/vip/vip9.sol:49]: will be uncommented once deployed

File: contracts/test/integration/vip/vip9.sol (hunk 3 above)
Comment (ElliotFriedman) [reply contracts/test/integration/vip/vip9.sol:36]: target is correct, description is not

**Review by eswak — DISMISSED** [review]
lgtm, hardhat validation is enough imo, you should pick a tool and either fully migrate your proposal flow to forge, or just write vip scripts in hardhat for now... also if volt starts to have a bunch of PCV deposits, you may want to setup a collateralization oracle like FEI, and keep it up to date in proposals, because it gets messy with assets spread over multiple contracts

**Review by k-xo — COMMENTED** [review]
LGTM pending deployments & uncommenting

File: contracts/test/integration/vip/vip9.sol [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        pure
+        override
+        returns (TimelockSimulation.action[] memory proposal)
+    {
+        proposal = new TimelockSimulation.action[](1);
+
+        address[] memory toWhitelist = new address[](3);
+        // toWhitelist[0] = address(daiDeposit);
+        // toWhitelist[1] = address(feiDeposit);
+        // toWhitelist[2] = address(usdcDeposit);
```
[the remark this answers is not in the document]
Comment (ElliotFriedman) [reply contracts/test/integration/vip/vip9.sol:34]: will fill once I've deployed the PCV Deposits

File: contracts/test/integration/vip/vip9.sol [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        // );
+        // assertEq(address(daiDeposit.core()), address(core));
+        // assertEq(address(feiDeposit.core()), address(core));
+        // assertEq(address(usdcDeposit.core()), address(core));
+        // assertEq(address(daiDeposit.cToken()), address(MainnetAddresses.CDAI));
+        // assertEq(address(feiDeposit.cToken()), address(MainnetAddresses.CFEI));
+        // assertEq(address(usdcDeposit.cToken()), address(MainnetAddresses.CUSDC));
+        // assertEq(address(daiDeposit.token()), address(MainnetAddresses.DAI));
+        // assertEq(address(feiDeposit.token()), address(MainnetAddresses.FEI));
+        // assertEq(address(usdcDeposit.token()), address(MainnetAddresses.USDC));
```
[the remark this answers is not in the document]
Comment (ElliotFriedman) [reply contracts/test/integration/vip/vip9.sol:66]: will uncomment once I've deployed the PCV Deposits

File: contracts/test/integration/vip/vip9.sol (hunk 3 above)
Comment (ElliotFriedman) [reply contracts/test/integration/vip/vip9.sol:36]: description is updated and target is correct!

File: proposals/dao/vip_9.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+const deploy: DeployUpgradeFunc = async (deployAddress: string, addresses: NamedAddresses, logging: boolean) => {
+  const erc20CompoundPCVDepositFactory = await ethers.getContractFactory('ERC20CompoundPCVDeposit');
+
+  const daiCompoundPCVDeposit = await erc20CompoundPCVDepositFactory.deploy(addresses.core, addresses.cDai);
+  const feiCompoundPCVDeposit = await erc20CompoundPCVDepositFactory.deploy(addresses.core, addresses.cFei);
+  const usdcCompoundPCVDeposit = await erc20CompoundPCVDepositFactory.deploy(addresses.core, addresses.cUsdc);
+
+  await daiCompoundPCVDeposit.deployed();
+  await feiCompoundPCVDeposit.deployed();
+  await usdcCompoundPCVDeposit.deployed();
```
Comment (ElliotFriedman) [inline proposals/dao/vip_9.ts:37]: wait on each deployment before going to the next one to prevent nonce mismatches and script errors on mainnet

**Review by thomas-waite — APPROVED** [review]
Verified deployed contracts and deploy params

**Review by eswak — APPROVED** [review]
deployed contracts look fine

**Review by Namaskar-1F64F — COMMENTED** [review]
looks good to me

File: protocol-configuration/mainnetAddresses.ts [hunk 7]
```diff
@@ -76,21 +76,21 @@ const MainnetAddresses: MainnetAddresses = {
     category: AddressCategory.Core,
     network: Network.Mainnet
   },
-  voltFusePCVDeposit: {
-    address: '0xFeBDf448C8484834bb399d930d7E1bdC773E23bA',
+  daiCompoundPCVDeposit: {
+    address: '0xE3cbfd618463B7198fa0743AbFA56170557cc880',
     artifactName: 'ERC20CompoundPCVDeposit',
     category: AddressCategory.Deprecated,
```
[the remark this answers is not in the document]
Comment (ElliotFriedman) [reply protocol-configuration/mainnetAddresses.ts:82]: we removed all PCV VOLT from this deposit, so at this point we don't think it makes sense to remove this deprecated contract

[the remark this answers is not in the document]
Comment (ElliotFriedman) [reply protocol-configuration/mainnetAddresses.ts:82]: category does need to be updated to PCV instead of deprecated

**Pull request merged**